flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/SinkFunction.java

```diff
@@ -35,6 +35,39 @@
 	 *
 	 * @param value The input record.
 	 * @throws Exception
+	 * @deprecated Use {@link #invoke(SinkContext, Object)}.
 	 */
-	void invoke(IN value) throws Exception;
+	@Deprecated
+	default void invoke(IN value) throws Exception {
+	}
+
+	/**
+	 * Writes the given value to the sink. This function is called for every record.
+	 *
+	 * @param context Additional context about the input record.
+	 * @param value The input record.
+	 * @throws Exception
+	 */
+	default void invoke(SinkContext context, IN value) throws Exception {
```

Comment:
Consider re-ordering the two arguments to more closely match the similar signature in `ProcessFunction`.

```
default void invoke(IN value, SinkContext ctx) throws Exception
```

Reply:
Makes sense, I will change this. 👍